plugins/modules/source_control/gitlab/gitlab_group_variable.py

```diff
@@ -56,6 +58,46 @@
         See GitLab documentation on acceptable values for a masked variable (U(https://docs.gitlab.com/ce/ci/variables/#masked-variables)).
     default: {}
     type: dict
+  variables:
+    version_added: 4.5.0
+    description:
+      - A list of dictionaries that represents CI/CD variables.
+      - This modules works internal with this sructure, even if the older I(vars) parameter is used.
+    default: []
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - The name of the variable
+        type: str
+        required: true
+      value:
+        description:
+          - The variable value
```

Comment:
We need documentation that `value` is required when state is present and is not required when state is absent.

Reply:
Yes. But I cannot figure out the the use of require_if based on a upper attribute (state: present)
This does not work for me.
```
value=dict(type='str', required_if=[ ('state', 'present', ('value')) ], no_log=True),
```
Any idea?
If not: sounds good to use required: False on all cases and document that value is required for state: present?

Reply:
yes. it is not possible. suboptions are it's own argument spec, so you can use `required_if` only on items ob suboption `variables`.

Reply:
So, what do you prefer?. Now value is required and it is similar as gitlab_project_variables.

1. We can leave it as is. And work in a fix on a next PR for both modules.
2. I can put required = False on all cases of atribute value and document that is required only for state: present.

Reply:
I prepared already a PR to fix after you've mentioned it https://github.com/ansible-collections/community.general/pull/4150  
When @felixfontein also agreed with it, I suggest so apply this also for gitlab_group_variable. Otherwise we'll introduce a bug in `variables` parameter, that affected not the older `vars` parameter.

Reply:
I just merged that PR.